### DISCUSSION TITLE
How to enable Language Auto Detection for Real Time Streaming?

### DISCUSSION THREAD

**Discussion by saxena666** [deepgram/community#602] [General help] [answered]
Looking to enable Language Auto Detection for Real Time Streaming. I am unable to find any help document. Could someone please help

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jkroll-deepgram** [comment, marked as answer]
Hi @sabysaxena, currently we only support [language detection](https://developers.deepgram.com/docs/language-detection) for pre-recorded audio, not for streaming. However, we have had customers send the first ~30 seconds of an audio stream to our pre-recorded language detection, and then transcribing the rest of the stream in that detected language.

**Reply by Sabysx** [reply]
Thanks. What about is the Language is mix of English plus something. Most conversations are never purely in a single language. 

How do you recommend to handle such conversations in real time streaming? 

Also, as the playground doesn't support testing real time streaming of languages, do you have other options to test this? This is very critical for the selection of Deepgram over other providers. Thanks

**Reply by jkroll-deepgram** [reply]
All of our ASR models are currently single-language (both batch and streaming), though our non-English models can generally also transcribe words and names in English.

I recommend using our [live streaming starter kit](https://github.com/deepgram/live-streaming-starter-kit/tree/main) for testing streaming. You can stream live audio from your mic, or stream a WAV file. The documentation for using the repo is here: https://developers.deepgram.com/docs/getting-started-with-the-streaming-test-suite
